src/v/raft/consensus.cc

```diff
@@ -2691,6 +2691,25 @@ ss::future<consensus::flushed> consensus::flush_log() {
 
     _flushed_offset = std::max(flushed_up_to, _flushed_offset);
     vlog(_ctxlog.trace, "flushed offset updated: {}", _flushed_offset);
+
+    const auto& conf = _configuration_manager.get_latest().current_config();
+    if (conf.voters.size() == 1) {
+        // single participant raft group, ensuring indices advance
+        auto dirty_offset = _log->offsets().dirty_offset;
+        process_append_entries_reply(
+          _self.id(),
+          append_entries_reply{
+            .target_node_id = _self,
+            .node_id = _self,
+            .group = _group,
+            .term = _term,
+            .last_flushed_log_index = _flushed_offset,
+            .last_dirty_log_index = dirty_offset,
+            .result = reply_result::success},
+          follower_req_seq{0},
+          dirty_offset);
+    }
```

Comment:
we were chatting about making this part of `replicate_stm`. In this form it is incorrect, we can not simply check the voter count in current configuration.